### ISSUE TITLE
Adding Legality check for every Gen (2 to 5)

### ISSUE THREAD

**Issue by AngelSora** [kwsch/PKHeX#846]
For example. I transfer a Lucario from Pokemon Platinum to Sun but this doesn't have legality check. Every gen should have Legality check, even if there is NO online for it.

**Comment by egzonqj** [comment]
That's a lot of work to be honest...

**Comment by AngelSora** [comment]
No matter how long this will be, i REALLY want it to be added.

**Comment by egzonqj** [comment]
Lot of people wants it, yeah

**Comment by kwsch** [comment]
Refer to https://github.com/kwsch/PKHeX/pull/190 and https://github.com/kwsch/PKHeX/issues/228.

4 months later -- the legality checking core `Core.cs` has been refactored to support multiple generations but the checking logic `Checks.cs`needs to be reorganized/broken up in order to check across generations more easily. Gathering encounter tables, learnsets, various tables (see `Tables#.cs`) and fine tuning the flow of logic is a huge time sink.

Gen 1 legality checking is the next realistic target. Imports are possible directly to Generation 7, and the checks required would be simple in comparison to the other generations. If any legality checking is added for past generations, it'd start with RBY.

tl;dr - large time investment required; less useful than $currentGen checking. It's on the to-do list...

**Comment by Mrtops** [comment]
I agree with kwsch that is a major time investment and therefore will be implemented over time.

**Comment by kwsch** [comment]
Closing, refer to latest commits.

**Comment by AngelSora** [comment]
I can't wait for the next Update !
